drivers/gpio/gpio_pca_series.c

```diff
@@ -0,0 +1,2178 @@
+							 (uint8_t *)&int_mask);
+		if (ret) {
+			goto out;
+		}
+	}
+
+	if (mode == GPIO_INT_MODE_DISABLED) {
+		(*int_fall) &= ~BIT(pin);
+		(*int_rise) &= ~BIT(pin);
+		if (cfg->part_cfg->flags & PCA_HAS_INT_MASK) {
+			int_mask |= BIT(pin); /** set 1 to disable interrupt */
+		}
+	} else {
+		if (trig == GPIO_INT_TRIG_BOTH) {
+			(*int_fall) |= BIT(pin);
+			(*int_rise) |= BIT(pin);
+		} else if (trig == GPIO_INT_TRIG_LOW) {
+			(*int_fall) |= BIT(pin);
+			(*int_rise) &= ~BIT(pin);
+		} else if (trig == GPIO_INT_TRIG_HIGH) {
+			(*int_fall) &= ~BIT(pin);
+			(*int_rise) |= BIT(pin);
+		}
+		if (cfg->part_cfg->flags & PCA_HAS_INT_MASK) {
+			int_mask &= ~BIT(pin); /** set 0 to enable interrupt */
+		}
+	}
+
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+	/** merge two buffer into low address */
+	gpio_pca_series_cache_2b_merge(dev, buf);
+	/** save config to cache even if this register is not present on device */
+	ret = gpio_pca_series_reg_update_cache(dev, PCA_REG_TYPE_2B_INTERRUPT_EDGE,
+						   (uint8_t *)&buf);
+	if (ret) {
+		LOG_ERR("get current interrupt config fail [%d]", ret);
+		goto out;
+	}
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+
+	/** enable latch if available, so we do not lost interrupt */
+	if (cfg->part_cfg->flags & PCA_HAS_LATCH) {
+		input_latch = ~int_mask;
+		ret = gpio_pca_series_reg_write(dev, PCA_REG_TYPE_1B_INPUT_LATCH,
+						(uint8_t *)&input_latch);
+		if (ret) {
+			goto out;
+		}
+	}
+	/** update interrupt mask register if available */
+	if (cfg->part_cfg->flags & PCA_HAS_INT_MASK) {
+		ret = gpio_pca_series_reg_write(dev, PCA_REG_TYPE_1B_INTERRUPT_MASK,
+						(uint8_t *)&int_mask);
+		if (ret) {
+			goto out;
+		}
+	}
+
+out:
+	k_sem_give(&data->lock);
+	return ret;
+}
+
+/**
+ * @brief Configure interrupt for device with hardware-selected interrupt edge
+ *
+ * @note This version is used by devices that have interrupt edge config
+ *       (aka PCA_HAS_INT_EXTEND), so interrupt only triggers on selected edge.
+ *       This applies to all pcal65xx devices.
+ *       This will configure interrupt mask register and interrupt edge register.
+ *       (All devices that have PCA_HAS_INT_EXTEND flag should have PCA_HAS_INT_MASK
+ *       flag. Otherwise, throw out error.)
+ *
+ * @param dev
+ * @param pin
+ * @param mode
+ * @param trig
+ * @return int
+ */
+static int gpio_pca_series_pin_interrupt_configure_extended(
+		const struct device *dev,
+		gpio_pin_t pin, enum gpio_int_mode mode,
+		enum gpio_int_trig trig)
+{
+	const struct gpio_pca_series_config *cfg = dev->config;
+	struct gpio_pca_series_data *data = dev->data;
+	uint64_t int_edge;
+	uint32_t int_mask, input_latch;
+	int ret;
+	uint32_t edge_cfg_shift = pin << 1U;
+	uint64_t edge_cfg_mask = 0b11 << edge_cfg_shift;
+
+	if (cfg->gpio_int.port == NULL) {
+		return -ENOTSUP;
+	}
+	/* Device does not support level-triggered interrupts. */
+	if (mode == GPIO_INT_MODE_LEVEL) {
+		return -ENOTSUP;
+	}
+#ifdef DEBUG
```

Comment:
I think we should probably create some form of Kconfig to control this behavior- We could also call the `gpio_pca_series_dump` function here, if we want to keep it present

Reply:
These code are meant for future developers: Checking feature flags in "extended" variant is only needed when developer wants to add new compatible device to this driver. For normal user, the api struct should've guaranteed that these flags are set, thus these code should be invisible to them. So it might be confusing if special KConfig is applied -- they are not supposed to see such err.

Btw, for completeness, I intended to add more checks in similar form. Please shed some light on how to properly handle this.

Reply:
I think adding a Kconfig to enable developer debugging options here makes sense, right? This way, developers can turn it on when they are working on the driver, but the Kconfig defaults to disabled, so a user won't normally see the error

Reply:
In Linux, it seems that driver-specific macros (which would not be exposed via Kconfig) are used: https://github.com/torvalds/linux/blob/98b1cc82c4affc16f5598d4fa14b1858671b2263/drivers/video/fbdev/cirrusfb.c#L1262. So I'm ok if we do something like this here, like `#ifdef NXP_PCA_DEBUG`

Reply:
> In Linux, it seems that driver-specific macros (which would not be exposed via Kconfig) are used: https://github.com/torvalds/linux/blob/98b1cc82c4affc16f5598d4fa14b1858671b2263/drivers/video/fbdev/cirrusfb.c#L1262. So I'm ok if we do something like this here, like #ifdef NXP_PCA_DEBUG

Then let's go this way : )